week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /*
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(destination => destination.distanceKms < 500)
+  .map(destination => destination.destinationName); // Complete here
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+var destinationNameReachableByFerry = travelDestinations
+  .filter(destination => destination.transportations.includes("ferry"))
+  .map(reachebleByFerry => reachebleByFerry.destinationName); // Complete here
 
+var destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(
+    destination =>
+      destination.transportations.includes("train") &&
+      destination.distanceKms > 300
+  )
+  .map(destination => destination.destinationName); // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
 
+destinationNamesMoreThan300KmsAwayByTrain;
```

Comment:
Why the variable name here again? Are you confusing this with calling functions? You don't need to do this with variables.

Reply:
to be honest i don't know why it is there. But maybe I wasn't attentive enough and i didn't notice it. I deleted it.